tests/components/london_air/test_sensor.py

```diff
@@ -1,34 +1,80 @@
-"""The tests for the tube_state platform."""
-import unittest
+"""The tests for the london_air platform."""
+import json
 
-import requests_mock
+from homeassistant.components.london_air.sensor import CONF_LOCATIONS
+from homeassistant.const import HTTP_OK, HTTP_SERVICE_UNAVAILABLE
+from homeassistant.setup import async_setup_component
 
-from homeassistant.components.london_air.sensor import CONF_LOCATIONS, URL
-from homeassistant.setup import setup_component
+from tests.async_mock import patch
+from tests.common import load_fixture
 
-from tests.common import get_test_home_assistant, load_fixture
+VALID_CONFIG = {"sensor": {"platform": "london_air", CONF_LOCATIONS: ["Merton"]}}
 
-VALID_CONFIG = {"platform": "london_air", CONF_LOCATIONS: ["Merton"]}
 
+class MockResponse:
+    """Class to represent a mocked response."""
 
-class TestLondonAirSensor(unittest.TestCase):
-    """Test the tube_state platform."""
+    def __init__(self, json_data, status_code):
+        """Initialize the mock response class."""
+        self.json_data = json_data
+        self.status_code = status_code
 
-    def setUp(self):
-        """Initialize values for this testcase class."""
-        self.hass = get_test_home_assistant()
-        self.config = VALID_CONFIG
-        self.addCleanup(self.hass.stop)
+    def json(self):
+        """Return the json of the response."""
+        return self.json_data
 
-    @requests_mock.Mocker()
-    def test_setup(self, mock_req):
-        """Test for operational tube_state sensor with proper attributes."""
-        mock_req.get(URL, text=load_fixture("london_air.json"))
-        assert setup_component(self.hass, "sensor", {"sensor": self.config})
-        self.hass.block_till_done()
 
-        state = self.hass.states.get("sensor.merton")
+def mocked_requests_get(*args, **kwargs):
+    """Mock requests.get invocations."""
+    url = str(args[0])
+
+    if "GroupName=London/Json" in url:
+        return MockResponse(json.loads(load_fixture("london_air.json")), HTTP_OK)
+
+
+def mocked_requests_get_failed(*args, **kwargs):
+    """Mock requests.get failure."""
+    return MockResponse({}, status_code=HTTP_SERVICE_UNAVAILABLE)
+
+
+async def test_valid_state(hass):
+    """Test for operational london_air sensor with proper attributes."""
+    with patch("requests.get", mocked_requests_get):
```

Comment:
Instead of mocking requests directly, use `requests_mock`:

```suggestion
async def test_valid_state(hass, requests_mock):
    """Test for operational london_air sensor with proper attributes."""
    requests_mock.get(url, text=load_fixture("london_air.json")
```